docs/guides/hardhat-walkthrough/hardhat-base-setup.md

```diff
@@ -0,0 +1,74 @@
+---
+sidebar_label: Hardhat basic setup
+sidebar_position: 1
+---
+
+# Setup your Hardhat project
+
+In this article will guide you through the process of seting up an Hardhat installation (using TypeScript), adding the [`@lukso/lsp-smart-contracts`](https://www.npmjs.com/package/@lukso/lsp-smart-contracts) package, creating a basic LSP7 contract, and deploying it on [LUKSO Testnet](http://docs.lukso.tech/networks/testnet/parameters).
+
+## Create Hardhat project
+
+The first thing to do is to create a new Hardhat project that will use TypeScript:
+
+```bash title="Setup new hardhat project"
+mkdir lukso-app
+cd lukso-app
+npx hardhat # select 'Create a TypeScript project' and use the default value for the rest of the setup
+```
+
+Once finished, you have a working hardhat setup!
+
+## Install packages & setup tools
+
+To work in the best condition possible, we will install libraries that includes tools, helpers and the [`@lukso/lsp-smart-contracts`](https://www.npmjs.com/package/@lukso/lsp-smart-contracts) package.
+
+### Install dependencies
+
+```js title="Install dependencies"
+npm i -D dotenv prettier
+npm i -s @lukso/lsp-smart-contracts
+```
+
+### Add a build script in your package.json
+
+Update your package.json with the following:
+
+```json title="Add a build script"
+"scripts": {
+  "build": "hardhat compile --force --show-stack-traces"
+},
+```
+
+### Create the prettier config
+
+Create a new file at the root of your project called `.prettierrc` with the following content:
+
+```json title="Create the prettier config"
+{
+  "trailingComma": "all",
+  "tabWidth": 2,
+  "semi": true,
+  "singleQuote": true,
+  "printWidth": 110
+}
+```
```

Comment:
is this important for this guide?

Reply:
It's important to have a good "complete" setup, which is why i put it here. So not mandatory, but definitely part of the dapp dev journey